osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs

```diff
@@ -73,10 +78,24 @@ public MultiSpectatorScreen(Room room, MultiplayerRoomUser[] users)
             instances = new PlayerArea[Users.Count];
         }
 
+        /// <summary>
+        /// Add a settings group to the bottom of the side container. Intended to be used by rulesets to add spectate-specific settings.
+        /// </summary>
+        /// <param name="settings">The settings group to be shown.</param>
+        public void AddSettingsAsync(PlayerSettingsGroup settings)
```

Comment:
Why is this async?

Reply:
When I originally tried adding the settings container from load there were problems with each `DrawableOsuRuleset` adding a container, even checking `SettingsAdded` in `MultiSpectatorScreen`, so I figured it needed to be added from `LoadComplete` and that I should also use `LoadComponenetAsync` in that case (maybe wrong); however, I can't call it from `DrawableOsuRuleset` because it needs to first verify via `SettingsAdded` that a settings container wasn't already added.

Reply:
> there were problems with each `DrawableOsuRuleset` adding a container

What problems specifically?

If those can be solved maybe this can happen synchronously and a whole bunch of bool-tracking badness could go away in favour of just a simple check.

Reply:
Actually, thinking about it again, I think I've suddenly realized why that was occurring... I'll try to put it back in load with synchronous loading

Reply:
My idea did not fix it... currently it looks like this in `DrawableOsuRuleset`:
```CSharp
if (multiSpectatorScreen == null)
    player.AddSettings(new ReplayAnalysisSettings(Config));
else if (!multiSpectatorScreen.SettingsAdded)
    multiSpectatorScreen.AddSettings(new ReplayAnalysisSettings(Config));
```
and `AddSettings` is basically the same, just using `Schedule` and no `LoadComponentAsync`

The problem is that multiple analysis settings get added to the screen.

Reply:
Using a lock fixes the problem